native/src/main/java/io/ballerina/stdlib/websocket/WebSocketResourceDispatcher.java

```diff
@@ -462,20 +469,49 @@ public static void dispatchOnText(WebSocketConnectionInfo connectionInfo, WebSoc
             if (finalFragment) {
                 stringAggregator.appendAggregateString(textMessage.getText());
                 int index = 0;
-                for (Type param : parameterTypes) {
-                    String typeName = param.getName();
-                    switch (typeName) {
-                    case WebSocketConstants.WEBSOCKET_CALLER:
-                        bValues[index++] = wsEndpoint;
-                        bValues[index++] = true;
-                        break;
-                    case WebSocketConstants.PARAM_TYPE_STRING:
-                        bValues[index++] = StringUtils.fromString(stringAggregator.getAggregateString());
-                        bValues[index++] = true;
-                        break;
-                    default:
-                        break;
+                try {
+                    for (Type param : parameterTypes) {
+                        int typeTag = param.getTag();
+                        switch (typeTag) {
+                            case OBJECT_TYPE_TAG:
+                                bValues[index++] = wsEndpoint;
+                                bValues[index++] = true;
```

Comment:
Use a single `bValues[index++] = true;` after L510 rather repeating it in all cases as it is common statement